### PR TITLE
[OPENTOK-29597] update OTPublisher initializer

### PR DESCRIPTION
**Pull request by Lucashuang0802**

### CODE REVIEW

File: 1.Hello-World/Hello-World/ViewController.m [hunk 1]
```diff
@@ -88,9 +88,9 @@ - (void)doConnect
  */
 - (void)doPublish
 {
-    _publisher =
-    [[OTPublisher alloc] initWithDelegate:self
-                                     name:[[UIDevice currentDevice] name]];
+    OTPublisherSettings *setting = [[OTPublisherSettings alloc] init];
```
Comment (jeffswartz) [inline 1.Hello-World/Hello-World/ViewController.m]: I recommend changing the name of this property to `_settings` (with an s) or maybe `settings`.

Comment (jeffswartz) [inline 1.Hello-World/Hello-World/ViewController.m]: And make this change throughout.

File: 8.Audio-Levels/Audio-Levels/TBViewController.m [hunk 2]
```diff
@@ -138,8 +138,9 @@ - (void)setupSession
 - (void)setupPublisher
 {
     // create one time publisher and style publisher
-	 _publisher = [[OTPublisher alloc]
-                              initWithDelegate:self name:self.publisherName];
+    OTPublisherSettings *setting = [[OTPublisherSettings alloc] init];
+    setting.name = self.publisherName;
+	 _publisher = [[OTPublisher alloc] initWithDelegate:self settings:setting];
```
Comment (jeffswartz) [inline 8.Audio-Levels/Audio-Levels/TBViewController.m]: The indentation of the previous three lines should match.

File: 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m [hunk 3]
```diff
@@ -19,14 +19,14 @@ @implementation TBExamplePublisher {
 #pragma mark - Object Lifecycle
 
 - (id)init {
-    self = [self initWithDelegate:nil name:nil];
+    self = [self initWithDelegate:nil settings:nil];
```
Comment (jeffswartz) [inline 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m]: Is it OK to pass in `nil` here? Or will you get a warning?

File: 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m [hunk 4]
```diff
@@ -19,14 +19,14 @@ @implementation TBExamplePublisher {
...
     if (self) {
         // nothing to do!
     }
     return self;
 }
 - (id)initWithDelegate:(id<OTPublisherDelegate>)delegate {
-    self = [self initWithDelegate:delegate name:nil];
+    self = [self initWithDelegate:delegate settings:nil];
```
Comment (jeffswartz) [inline 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m]: Is it OK to pass in `nil` here? Or will you get a warning?

File: 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m (hunk 3 above)
Comment (Lucashuang0802) [inline 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m]: Yes, you will have a warning.

File: 1.Hello-World/Hello-World/ViewController.m (hunk 1 above)
Comment (Lucashuang0802) [inline 1.Hello-World/Hello-World/ViewController.m]: 👍

File: 8.Audio-Levels/Audio-Levels/TBViewController.m (hunk 2 above)
Comment (Lucashuang0802) [inline 8.Audio-Levels/Audio-Levels/TBViewController.m]: good catch. Thanks

File: 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m (hunk 4 above)
Comment (Lucashuang0802) [inline 2.Custom-Video-Driver/Lets-Build-OTPublisher/TBExamplePublisher.m]: Yes, you will have a warning.

File: 8.Audio-Levels/Audio-Levels/TBViewController.m (hunk 2 above)
Comment (Lucashuang0802) [inline 8.Audio-Levels/Audio-Levels/TBViewController.m]: 👍

**Pull request merged**